src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2803,5 +2803,11 @@ public ImmutableArray<AssemblyIdentity> GetUnreferencedAssemblyIdentities(Diagno
         }
 
         internal abstract bool IsUnreferencedAssemblyIdentityDiagnosticCode(int code);
+
+        /// <summary>
+        /// Returns the required language version found in a <see cref="Diagnostic"/>, if any is found.
+        /// Returns null if none is found.
+        /// </summary>
+        public abstract string GetRequiredLanguageVersion(Diagnostic diagnostic);
```

Comment:
>public abstract string GetRequiredLanguageVersion(Diagnostic diagnostic); [](start = 8, length = 73)

It feels strange that we require one to have a compilation instance in order to get this information, whereas the diagnostics instance is all we need.